libraries/extractor/src/main/java/androidx/media3/extractor/ts/H265Reader.java

```diff
@@ -374,6 +377,17 @@ public void endNalUnit(long position, int offset, boolean hasOutputFormat) {
       }
     }
 
+    public void end(long position) {
+      sampleIsKeyframe = nalUnitHasKeyframeData;
+      // Output a sample with the NAL units since the current nalUnitPosition
+      outputSample(/* offset= */ (int) (position - nalUnitPosition));
+      // Output a final sample with the remaining NAL units up to the passed position
+      samplePosition = nalUnitPosition;
+      nalUnitPosition = position;
+      outputSample(/* offset= */ 0);
```

Comment:
@dsparano, why are we outputting two different samples with the same timestamp for H.265 and not other readers? Please see the H.265 specific dump files in this PR. 

This may cause some problem with the renderers as it takes into consideration the last output time and only reads more samples when its timestamp is greater than the previous one.

Reply:
I did it like that because with HEVC at the boundary of HLS segments the sample taken up to the `position` was giving a merged Access Unit (AU), if compared with the main branch, of the last AU of the previous segment and the first one of the next segment. With the code you see now it returns the same AUs compared with main code across boundaries but also the very last AU of the last segment of a VOD HLS, which main fails to provide.

The challenge here is that the end of input at segment boundaries is not so much different than the end of input at end of file or VOD HLS, so I couldn't find a better way to handle them both without re-designing the entire logic more heavily.

If you ask why this is not needed for H264 and the other readers I think it is because of the different rules in determining the boundaries of AUs between the various codecs.

Any design tweak you can suggest I'm happy to give it a try.

Reply:
will you be able to provide an example stream which causes problems without this splitting?